### PR TITLE
PRO-4392: Tasklist Bug with Executors Roles

### PR DESCRIPTION
**Pull request by LucaDelBuonoHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRO-4392

### Change description ###
Fix bug with Executor Roles question being marked as complete when an executor renounces but Executor Notified question still not answered, which leads to Tasklist marking Executors task as completed

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: app/steps/ui/executors/roles/index.js [hunk 1]
```diff
@@ -40,7 +40,16 @@ class ExecutorRoles extends CollectionStep {
...
+        return [every(ctx.list, exec => {
+            return exec.isApplying ||
+                (
+                    !isEmpty(exec.notApplyingReason) &&
+                    (
+                        (exec.notApplyingReason === json.optionPowerReserved && !isEmpty(exec.executorNotified)) ||
+                        (exec.notApplyingReason !== json.optionPowerReserved)
+                    )
+                );
+        }), 'inProgress'];
```
Comment (shepda) [inline app/steps/ui/executors/roles/index.js]: Please add unit tests for this logic

Comment (LucaDelBuonoHMCTS) [reply app/steps/ui/executors/roles/index.js]: Added

File: test/unit/testExecutorRoles.js [hunk 2]
```diff
@@ -63,6 +63,132 @@ describe('ExecutorRoles', () => {
         });
     });
 
+    describe('isComplete()', () => {
+        it('should return the correct step completion status status when all executors are applying', (done) => {
```
Comment (shepda) [inline test/unit/testExecutorRoles.js]: Please remove the duplicated 'status' from the description in these tests

File: test/unit/testExecutorRoles.js [hunk 3]
```diff
@@ -63,6 +63,132 @@ describe('ExecutorRoles', () => {
...
+            const ctx = {
+                'list': [
```
Comment (shepda) [inline test/unit/testExecutorRoles.js]: Please remove the quotes from object params in these tests to keep consistency throughout the file

Comment (LucaDelBuonoHMCTS) [reply test/unit/testExecutorRoles.js]: Ha good spot, I actually pasted it from some other file so will search for that and remove the quotes from there as well.

Comment (LucaDelBuonoHMCTS) [reply test/unit/testExecutorRoles.js]: I found the same all over the place... removing them all.

**Pull request merged**